### PR TITLE
CODE-23 configure prettier to have no semi colons and auto fix no semi colon errors

### PR DESCRIPTION
**Pull request by NagaPradeepGajula-osi**
## Description
In an effort towards common standards across the digital services teams we are trying to consolidate some of the best practices from each team. Opting no semi colon is one of them.

## Tests

- [ ] I used TDD to develop the feature
- [ ] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests
No tests were included as this is a just linting configuration for which we don't have tests.

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I ran all my tests locally which were green.
- [ ] My code adds no new issues to Code Climate
- [ ] I ran all the linters for the project.
- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [ ] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

**Comment by wmitchell-cambria** [comment]
Were there any manual fixes you had to make, or was this entirely a `--fix` operation?

**Comment by NagaPradeepGajula-osi** [comment]
no manual changes just using `--fix`

File: app/javascript/Application/components/Assessment/Assessment.test.js [hunk 1]
```diff
@@ -1,249 +1,245 @@
...
-          const wrapper = shallow(
-            <Assessment onAssessmentUpdate={mockFn} assessment={initialAssessment} i18n={i18n} />
-          );
-          const instance = wrapper.instance();
+          const initialAssessment = clone(assessment)
+          initialAssessment.state.domains[0] = enhanceDomainToCaregiver(initialAssessment.state.domains[0])
+          initialAssessment.state.domains.push({ ...initialAssessment.state.domains[0], caregiver_index: 'b' })
+          initialAssessment.state.domains.push({ ...initialAssessment.state.domains[0], caregiver_index: 'c' })
+          const mockFn = jest.fn()
+          const wrapper = shallow(<Assessment onAssessmentUpdate={mockFn} assessment={initialAssessment} i18n={i18n} />)
```
Comment (wmitchell-cambria) [inline app/javascript/Application/components/Assessment/Assessment.test.js]: The handful of differences in additions and deletions seem to just be these types of lines that are right on the cusp of 120 characters in length.

**Comment by NagaPradeepGajula-osi** [comment]
@denys-davydov there was a discussion about semi-colons between all DS team leads and the verdict out of it was to have them removed.

**Comment by wmitchell-cambria** [comment]
@denys-davydov  - There was an informal discussion between front-end leads across teams (specifically @edgibbs and @ratneshraval). In the end, I think this is within the realm of Code Quality to decide on shared standards. If Pradeep had decided to go the other way and update Intake and CALS to use semicolons, I would defer to him and approve that as well.

**Comment by ratneshraval** [comment]
@NagaPradeepGajula-osi 
I think I suggested to have semi-colon even though I personally don't like them. Most of the modern lint/ide will automatically insert them or remove them for you so in the end one way or another doesn't really make a difference to me.

There was more discussion before on this over here https://github.com/ca-cwds/react-wood-duck/pull/71

**Comment by NagaPradeepGajula-osi** [comment]
Thanks for adding in more context @ratneshraval

**Comment by denys-davydov** [comment]
@ratneshraval @wmitchell-cambria @NagaPradeepGajula-osi Guys, thank you for more context on this. I look into the discussion in the above-mentioned PR https://github.com/ca-cwds/react-wood-duck/pull/71 and I can see that the decision was made to **have** semicolons and the PR was closed.

**Comment by NagaPradeepGajula-osi** [comment]
@denys-davydov there was no verdict out of the PR to keep the semicolons. It was just closed without a decision and if you look at the PR clearly most of them voted for to remove them. This Pr was just an additional context but, as Will mentioned there was an informal discussion later and agreed on removing them.

**Comment by wmitchell-cambria** [comment]
Looping in @eckdanny-osi as well, since there seems to be some disagreement here. My reading of the linked react-wood-duck issue is not that the decision was made, but rather that no decision was reach one way or the other, and Danny closed the issue.

In case it comes up, I want to point out that ASI is not a serious concern for us. Prettier did not support `semicolons: false` originally because it was harder to implement internally, not because the code itself was ambiguous. Prettier today handles all the cases where not having a semicolon may introduce an ASI failure, and combined with the Javascript processing we already do for ES6 features, the chance of a bug due to not having semicolons is approximately zero.

That being said, I believe the ownership of this decision lies with the Code Quality team. Building consensus on any nigh-religious issue like this is impossible, and we need some way to make decisions. Empowering that team to make the call is the way to do that.

**Comment by edgibbs** [comment]
OK, so I'll weigh in on the semicolon debate.  I don't have a strong opinion either way, but we're currently running it differently in different code bases.  Up until more recently, I'd say the edge cases in Javascript with leaving off semicolons concerned me, but ES6+ and Babel have pretty much eliminated this worry. Prettier takes care of this for me either way.  We do need to pick one way of doing it and if the Code Quality team has decided on no semicolons I think that's fine.

**Comment by eckdanny-osi** [comment]
I recall a windy debate on this subject _already having been settled_. (in [intake](https://github.com/ca-cwds/intake)?)

😡GitHub issues were disabled so I can't retrieve the old thread. Is the same debate happening again now with no historical reference?

**Comment by wmitchell-cambria** [comment]
Hmm, Intake has been no-semicolons at least since March, which is when I started. I'm not aware of any discussions around it since then. It's possible more discussions were had with the previous dev team. When the new Intake front-end devs came on board, we did a poll and the majority were in favor of sticking with no-semicolons.

**Comment by NagaPradeepGajula-osi** [comment]
There were no semicolons earlier also since Nov, 2017 when I started working in intake.

**Comment by eckdanny-osi** [comment]
This bike shedding. I feel much stronger about _not_ having this debate than I do about pretty printing config. If the the program wants to mandate, we can roll with it.

@denys-davydov's suprise ([above](https://github.com/ca-cwds/cans/pull/223#pullrequestreview-166302602)) is bit concerning... in the future, perhaps we can avoid blindsiding the [leading contributors of a repo](https://github.com/ca-cwds/cans/graphs/contributors?from=2018-07-15&to=2018-10-15&type=c) with decisions made outside their knowledge?

**Comment by denys-davydov** [comment]
So I have here 5 js projects I touched in one way or another since Oct, 2017:
https://github.com/ca-cwds/case-management
https://github.com/ca-cwds/dashboard
https://github.com/ca-cwds/react-wood-duck
https://github.com/ca-cwds/design-system
and of course
https://github.com/ca-cwds/cans

All of them have semicolons. Sorry, never touched intake or cals.
So I'm surprised again to see so many no-semicolon believers here :) 

@NagaPradeepGajula-osi please don't forget to update the mentioned projects.

**Comment by NagaPradeepGajula-osi** [comment]
@denys-davydov we are going to update all of the DS in near future. This would also be applied to County-admin right after CANS. Thanks for the review.

**Comment by NagaPradeepGajula-osi** [comment]
for future reference there were no errors or warnings after the changes while running the lint.
screen shot 2018-10-19 at 4 40 50 pm

**Pull request merged**